### PR TITLE
Fix bug: LobbyScreen UI has a bug on the Difficulty toggle group.

### PR DESCRIPTION
**Pull request by Toneyisnow**
Bug: https://bugs.vcmi.eu/view.php?id=3102

Scenario: In LoadGame Lobby screen, the difficulty toggle group is showing multiple choices, but actually only one valid is working.

The reason is, in the Lobby screen Initialization code will set the difficulty = 0, and each time the player changes a map in the selection, the toggle group control is not resetting the difficulty buttons to disabled state.

How fix:
Add a new method to ToggleGroup class: setSelectedOnly, which will disable all other buttons and then set the selected button.

Note:
During the game loading time, the client is loading the map, and send a NetPack to server: LobbySetMap, and send a NetPack to Interface: LobbyUpdateState.
In the LobbyUpdateState it sets the map difficulty.

### CODE REVIEW

File: client/widgets/Buttons.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@
 #include "../windows/InfoWindows.h"
 #include "../../lib/CConfigHandler.h"
 
+using namespace std;
```
Comment (alexvins) [inline client/widgets/Buttons.cpp]: Global `using` is not recommended.

File: client/widgets/Buttons.cpp [hunk 2]
```diff
@@ -425,6 +437,17 @@ void CToggleGroup::setSelected(int id)
 	selectionChanged(id);
 }
 
+void CToggleGroup::setSelectedOnly(int id)
+{
+	for (map<int, shared_ptr<CToggleBase> >::iterator it = buttons.begin(); it != buttons.end(); it++)
```
Comment (alexvins) [inline client/widgets/Buttons.cpp]: No space after `for` please and use `auto` in iterator declaration.

File: client/widgets/Buttons.cpp (hunk 1 above)
Comment (ArseniyShestakov) [reply client/widgets/Buttons.cpp]: Tbh I think we had long talks about it on Slack so I strongly against ```using``` for something as common as ```std```.

Comment (Toneyisnow) [reply client/widgets/Buttons.cpp]: Ok, that's fine.

File: client/widgets/Buttons.cpp (hunk 2 above)
Comment (Toneyisnow) [reply client/widgets/Buttons.cpp]: Make sense.

**Pull request merged**